### PR TITLE
Fixed type checking errors in node.py

### LINKED ISSUE #67: [Fall 2021] Step 1: Pyre - Fix type checking errors in Pytorch `torch/fx/node.py`
This task is part of a bigger effort to migrate Pytorch from Mypy to Pyre. The goal of this task for you is to gain comfort with the Python type system and the types of errors Pyre gives. Future Pysa project work and debugging builds on this foundation.

## Development Environment Setup
- Set up [Pytorch for development](https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#developing-pytorch)  and run [Mypy](https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#running-mypy) by following their contribution guide

## Task Description
Fix the Pyre type checking errors in `torch/fx/node.py`:
```
torch/fx/node.py:363:20 Incompatible variable type [9]: placeholder_names is declared to have type `List[str]` but is used as type `None`.
torch/fx/node.py:364:20 Incompatible variable type [9]: maybe_return_typename is declared to have type `List[str]` but is used as type `None`.
```  

## Submitting a PR

We use the following linters internally, so to save everyone's time, please make sure you run the following linters locally and fix errors related to the files you modified before submitting a PR:
```
black && usort format . && flake8
```

To install the linters, you can run the following command:
```
pip install flake8 usort black==21.4b2
```

### PR DESCRIPTION
**Pull request by Onyiee**
Fixes [issue#67](https://github.com/MLH-Fellowship/pyre-check/issues/67)
This PR fixes the type checking errors in Pytorch torch/fx/node.py . 
The variable types in 363:20 and 364:20 were declared to have type `List[str]`  but were  assigned a value of  `None`. This caused an incompatitble variable type error.  I changed the type from `List[str]` to `Optional[List[str]` . This therefore fixed the incompatitble variable type error. 

Signed-off-by: Onyemowo  Agbo
@onionymous
@0xedward

### CODE REVIEW

File: torch/fx/node.py [hunk 1]
```diff
@@ -1,33 +1,52 @@
...
+from typing import TYPE_CHECKING, Union, Callable, Any, Tuple, List, Optional, Dict, Set
+
+import torch
+from torch.fx.operator_schemas import (
+    normalize_function,
+    normalize_module,
+    ArgsKwargsPair,
+)
+from ._compatibility import compatibility
+from .immutable_collections import immutable_dict, immutable_list
```
Comment (jamesr66a) [inline torch/fx/node.py:14]: If these changes are being done due to an automatic IDE formatting plugin, please disable that plugin and revert these changes. Extraneous formatting changes mess up the blame information in the source files and can make further development more difficult

**Review by jamesr66a — CHANGES_REQUESTED** [review]
Please disable auto-formatting and re-request review

File: torch/fx/node.py (hunk 1 above)
Comment (Onyiee) [reply torch/fx/node.py]: I have reverted the formatting changes

**Review by jamesr66a — APPROVED** [review]

**Comment by jamesr66a** [comment]
Thank you!

**Comment by jamesr66a** [comment]
Actually @Onyiee, one last thing before we can merge, the test `TestFXAPIBackwardCompatibility. test_function_back_compat` is failing because the signature of a publicly facing API is changing. Can you run that test with the `--accept` flag, add the files that get changed when you do that, then push those changes to the PR? Thanks
